neurodamus/gj_user_corrections.py

```diff
@@ -0,0 +1,181 @@
+# @file gj_user_corrections.py
+# @brief Script for loading user corrections on gap junction connectivity
+# @author Oren Amsalem, Weina Ji
+# @date 2024-09-09
+
+
+import logging
+import numpy as np
+import pickle
+from .core import MPI
+from .core import NeurodamusCore as Nd
+from .core.configuration import ConfigurationError, SimConfig
+
+non_stochastic_mechs = ['NaTs2_t', 'SKv3_1', 'Nap_Et2', 'Ih', 'Im', 'KdShu2007',
+                        'K_Pst', 'K_Tst', 'Ca', 'SK_E2', 'Ca_LVAst', 'CaDynamics_E2',
+                        'NaTa_t', 'CaDynamics_DC0', 'Ca_HVA2', 'NaTg'] + \
+                       ['TC_cad', 'TC_ih_Bud97', 'TC_Nap_Et2', 'TC_iA', 'TC_iL', 'TC_HH',
+                        'TC_iT_Des98'] + \
+                       ['kdrb', 'na3', 'kap', 'hd', 'can', 'cal', 'cat', 'cagk', 'kca',
+                        'cacum', 'kdb', 'kmb', 'kad', 'nax', 'cacumb']
+
+stochastic_mechs = ['StochKv', 'StochKv2', 'StochKv3']
+
+
+def load_user_modifications(gj_manager):
+    node_manager = gj_manager.cell_manager
+    settings = SimConfig.beta_features
+    gjc = settings.get('gjc')
+
+    # deterministic_StochKv
+    if settings.get('determanisitc_stoch'):
+        logging.info("Set deterministic = 1 for StochKv")
+        _determanisitc_stoch(node_manager)
+
+    # update gap conductance
+    if settings.get('procedure_type') in ['validation_sim', 'find_holding_current']:
+        process_gap_conns = _update_conductance(gjc, gj_manager)
+        all_ranks_total = MPI.allreduce(process_gap_conns, MPI.SUM)
+        logging.info(f"Set GJc = {gjc} for {int(all_ranks_total)} gap synapses")
+
+    # remove active channels
+    remove_channels = settings.get('remove_channels')
+    Mechanisims = []
+    if remove_channels == 'all':
+        Mechanisims = non_stochastic_mechs + stochastic_mechs
+    if remove_channels == 'only_stoch':
+        Mechanisims = stochastic_mechs
+    if remove_channels == 'only_non_stoch':
+        Mechanisims = non_stochastic_mechs
+    if remove_channels:
+        logging.info("Removing channels type = " + remove_channels)
+        _perform_remove_channels(node_manager, Mechanisims)
```

Comment:
Unnecessary checks, var naming typos, missing warning
```suggestion
    if remove_channels:
        if remove_channels == 'all':
            rm_mechanisms = non_stochastic_mechs + stochastic_mechs
        elif remove_channels == 'only_stoch':
            rm_mechanisms = stochastic_mechs
        elif remove_channels == 'only_non_stoch':
            rm_mechanisms = non_stochastic_mechs
        else:
            logging.warning("Unknown GJ remove_channels setting: %s", remove_channels)
            rm_mechanisms = []
        logging.info("Removing channels type = " + remove_channels)
        _perform_remove_channels(node_manager, rm_mechanisms)
```

Reply:
Done!